### PR TITLE
Updated README.md with Introduction, Installation, and Usage portions.

### PR DESCRIPTION
**Pull request by p4rsec**

### CODE REVIEW

**Comment by p4rsec** [comment]
In regards to issue #75

File: README.md [hunk 1]
```diff
@@ -1,4 +1,38 @@
...
+### Table of Contents
+
+1. Installation
+2. Connecting to IRC
+
+***
+
+## Introduction
+
+Legobot is a platform that aims to make interactive IRC bots less of a chore and more of a simple pleasure. Legobot itself is a skeleton framework to be built upon; Legobot provides the basic connectivity and ability to listen to message object (called threads) and respond to users based on the content of the message. Legobot is extensible via writing simple functions called "Legos", which can be easily added, edited, modified, and removed based on the need. The bot provides an easy-to-use "message" object when needed, which eliminated the need to parse raw IRC strings; this allows faster, more efficient processing of message contents, and quicker response time. Currently, IRC is the bot's main function; however, the bot is easily extensible to work with other messaging applications, or whatever you might find a need for. Legobot itself is not an IRC bot, it is an extensible, easy to use bot framework for quick, painless deployment of different forms of chat bots (or other if you so choose).
```
Comment (bbriggs) [inline README.md]: Legobot is multi-protocol, not IRC.

Comment (bbriggs) [inline README.md]: Message object isn't called a thread. Each Lego and Message is spun up on a thread, however. That's not exposed to the user or developer.

Comment (bbriggs) [inline README.md]: Conflict of tense: "which eliminated"

Comment (bbriggs) [inline README.md]: This is a big, long run-on. Consider breaking into bullet-points per feature?

Comment (bbriggs) [inline README.md]: "Currently, IRC is the bot's main function"

Say, "The bot currently ships with a single built-in connector for IRC. Other connectors are easy to develop and more are on the roadmap for the near future" or something.

File: README.md [hunk 2]
```diff
@@ -1,4 +1,38 @@
...
+
+***
+
+## Installation
+
+Installation is quite simple. Just run `pip install Legobot`. You can also easily install this as a non-privileged user by running `pip install --user Legobot` instead (this is recommended).
+
+## Connecting to IRC
+
+Legobot is intended to be simple, flexible, and painless; thereby allowing the user to control all functionality. Consequently, Legobot doesn't really *do* much on its own, other than monitor IRC messages and manage connections. All triggers, responses, and functions are yours to create: this could range from an automated function to report Nginx monitoring info at a set interval, to a function for checking the weather, to a function that simply states "Hello, World!". The possibilities are endless, only bound by your imaginations (and will to create).
```
Comment (bbriggs) [inline README.md]: I should hire you to sell Apple devices

File: README.md [hunk 3]
```diff
@@ -1,4 +1,38 @@
...
+
+## Installation
+
+Installation is quite simple. Just run `pip install Legobot`. You can also easily install this as a non-privileged user by running `pip install --user Legobot` instead (this is recommended).
+
+## Connecting to IRC
+
+Legobot is intended to be simple, flexible, and painless; thereby allowing the user to control all functionality. Consequently, Legobot doesn't really *do* much on its own, other than monitor IRC messages and manage connections. All triggers, responses, and functions are yours to create: this could range from an automated function to report Nginx monitoring info at a set interval, to a function for checking the weather, to a function that simply states "Hello, World!". The possibilities are endless, only bound by your imaginations (and will to create). 
+
+To connect the bot to IRC, we must edit the config file. Here is an example using freenode:
```
Comment (bbriggs) [inline README.md]: s/edit/create and run an instance of the Legobot class/

File: README.md [hunk 4]
```diff
@@ -1,4 +1,38 @@
...
+
+```python
```
Comment (bbriggs) [inline README.md]: This whole block is going to change as we iron out the API. No worries.

File: README.md [hunk 5]
```diff
@@ -1,4 +1,38 @@
...
+import Legobot
+HOST = 'irc.freenode.com'
+PORT = 6667
+NICK = 'legobot'
+CHANS = [('#freenoode', ''), ('#bitcoin', ''), ('#legobot', '')] # List of tuples. Format is (channel, password)
```
Comment (bbriggs) [inline README.md]: Check the current Legobot.py in `develop` for how we instantiate the bot now. No more list of tuples! hooray!

File: README.md [hunk 6]
```diff
@@ -10,7 +10,21 @@
...
 
-Legobot is a platform that aims to make interactive IRC bots less of a chore and more of a simple pleasure. Legobot itself is a skeleton framework to be built upon; Legobot provides the basic connectivity and ability to listen to message object (called threads) and respond to users based on the content of the message. Legobot is extensible via writing simple functions called "Legos", which can be easily added, edited, modified, and removed based on the need. The bot provides an easy-to-use "message" object when needed, which eliminated the need to parse raw IRC strings; this allows faster, more efficient processing of message contents, and quicker response time. Currently, IRC is the bot's main function; however, the bot is easily extensible to work with other messaging applications, or whatever you might find a need for. Legobot itself is not an IRC bot, it is an extensible, easy to use bot framework for quick, painless deployment of different forms of chat bots (or other if you so choose). 
+Legobot is a platform that aims to make interactive multi-protocol bots less of a chore and more of a simple pleasure. 
+Legobot itself is: 
+..* a skeleton framework to be built upon
```
Comment (bbriggs) [inline README.md]: Use http://dillinger.io/ to check your .md as you write it

File: README.md [hunk 7]
```diff
@@ -20,17 +34,19 @@ Installation is quite simple. Just run `pip install Legobot`. You can also easil
 
 ## Connecting to IRC
 
-Legobot is intended to be simple, flexible, and painless; thereby allowing the user to control all functionality. Consequently, Legobot doesn't really *do* much on its own, other than monitor IRC messages and manage connections. All triggers, responses, and functions are yours to create: this could range from an automated function to report Nginx monitoring info at a set interval, to a function for checking the weather, to a function that simply states "Hello, World!". The possibilities are endless, only bound by your imaginations (and will to create). 
+Legobot is intended to be simple, flexible, and painless; thereby allowing the user to control all functionality. Consequently, Legobot doesn't really *do* much on its own, other than monitor IRC messages and manage connections. All triggers, responses, and functions are yours to create: this could range from an automated function to report Nginx monitoring info at a set interval, to a function for checking the weather, to a function that simply states "Hello, World!". The possibilities are endless, only bound by your imagination (and will to create).
```
Comment (bbriggs) [inline README.md]: s/IRC//

**Pull request merged**